src/ui/components/usersListTable/UsersListTable.tsx

```diff
@@ -0,0 +1,151 @@
+
+  return (
+    <div className='users-list-table-container'>
+      <table className='users-list-table'>
+        <thead>
+          <tr>
+            <th>User Info</th>
+            <th>Auth Method</th>
+            <th>Time joined</th>
+          </tr>
+        </thead>
+        <tbody className='text-small'>
+          {isLoading ? (
+            PlaceholderTableRows()
+          ) : displayedUsers.length > 0 ? (
+            displayedUsers.map((user, index) => UserTableRow({ user: user, index }))
+          ) : (
+            <tr>
+              <td>
+                <NoUsers />
+              </td>
+            </tr>
+          )}
+        </tbody>
+      </table>
+
+      <div className='user-list-footer'>
+        {UserListPagination({
+          ...props,
+          offset,
+          limit,
+        })}
+      </div>
+    </div>
+  )
+
+  function PlaceholderTableRows() {
+    return new Array(limit).fill(null).map((_, index) => (
+      <tr key={index} className='user-row placeholder'>
+        <td colSpan={3}>
+          <div className='user-info'></div>
+        </td>
+      </tr>
+    ))
+  }
+}
+
+const UserTableRow: React.FC<{ user: User; index?: number }> = (props) => {
+  const { user, index } = props
+  return (
+    <tr key={index} className='user-row'>
+      <td>{UserInfo(user)}</td>
+      <td>{UserRecipePill(user)}</td>
+      <td>{UserDate(user)}</td>
+    </tr>
+  )
+}
+
+const UserInfo = (user: User) => {
+  const { firstName, lastName, email, phoneNumber: phone } = user.user
+  const name = `${firstName ?? ''} ${lastName ?? ''}`.trim()
+  return (
+    <div className='user-info'>
+      <div className='main'>{name || email || (phone && PhoneDisplay(phone))}</div>
+      {email && name && <div>{email}</div>}
+      {phone && (name || email) && <div>{PhoneDisplay(phone)}</div>}
+    </div>
+  )
+}
+
+const UserRecipePill = (user: User) => {
+  const thirdpartyId = user.recipeId === UserRecipeType.thirdparty && user.user.thirdParty.id
+  return (
+    <div className={`pill ${user.recipeId} ${thirdpartyId}`}>
+      {UserRecipeTypeText[user.recipeId]}
+      {thirdpartyId && <span> - {thirdpartyId}</span>}
+    </div>
+  )
+}
+
+const UserDate = (user: User) => {
+  return <div className='user-date'>{user.user.timeJoined && formatLongDate(user.user.timeJoined)}</div>
+}
+
+const UserRecipeTypeText: Record<UserRecipeType, string> = {
+  [UserRecipeType.emailpassword]: 'Email password',
+  [UserRecipeType.passwordless]: 'Passwordless',
+  [UserRecipeType.thirdparty]: 'Third party',
+}
+
+const UserListPagination = (props: UserListProps) => {
+  const { limit, offset, goToNext, offsetChange, nextPaginationToken, isLoading, count } = {
+    offset: 0,
+    limit: LIST_DEFAULT_LIMIT,
+    ...props,
+  }
+  return (
+    <div className='users-list-pagination'>
+      <p className='users-list-pagination-count text-small'>
+        {offset + 1}- {offset + limit} of {count}
+      </p>
```

Comment:
Try to add comments to your code (in this and other places)

Reply:
I have break it down to smaller functions